File: output/StarCoder/avatar/Java/Python/atcoder_ABC150_C.py
import java.util.ArrayList ; import java.util.Arrays ; import java.util.List ; import java.util.Scanner ; def permutation(list, target, ans):
    if len(target) <= 1:
        list.add(ans + target)
    else:
        for i in range(len(target)):
            permutation(list, target[:i] + target[i+1:], ans + target[i])
    return list

def main():
    scanner = Scanner(System.in)
    n = Integer.parseInt(scanner.next())
    line = [""] * 2
    for i in range(2):
        for j in range(n):
            line[i] += scanner.next()
    number = ""
    for i in range(1, n+1):
        number += str(i)
    listA = ArrayList()
    permutation(listA, number, "")
    sum = 0
    for j in range(2):
        for i in range(listA.size()):
            if listA.get(i).equals(line[j]):
                if sum == 0:
                    sum += i
                else:
                    sum -= i
    System.out.println(Math.abs(sum))

if __name__ == "__main__":
    main()

File: output/StarCoder/avatar/Python/Java/atcoder_ABC150_C.java
import java.util.Scanner;
import java.util.ArrayList;
import java.util.List;
import java.util.Arrays;

public class atcoder_ABC150_C {

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int n = sc.nextInt();
        List<List<Integer>> orig = new ArrayList<List<Integer>>();
        for (int i = 0; i < n; i++) {
            List<Integer> l = new ArrayList<Integer>();
            for (int j = 0; j < n; j++) {
                l.add(j + 1);
            }
            orig.add(l);
        }
        int[] p = new int[n];
        int[] q = new int[n];
        for (int i = 0; i < n; i++) {
            p[i] = sc.nextInt();
            q[i] = sc.nextInt();
        }
        int pn = orig.indexOf(Arrays.asList(p));
        int qn = orig.indexOf(Arrays.asList(q));
        System.out.println(Math.abs(pn - qn));
    }
}